src/test/java/com/yelp/nrtsearch/server/field/PrefixFieldDefTest.java

```diff
@@ -0,0 +1,174 @@
+      assertEquals("min_chars [1] must be greater than zero", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testInvalidMaxChars() {
+    try {
+      Field field =
+          Field.newBuilder()
+              .setSearch(true)
+              .setIndexPrefixes(IndexPrefixes.newBuilder().setMinChars(2).setMaxChars(20).build())
+              .build();
+      createFieldDef(field);
+      fail("Should throw IllegalArgumentException for max_chars >= 20");
+    } catch (IllegalArgumentException e) {
+      assertEquals("max_chars [20] must be less than 20", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testInvalidMinMaxChars() {
+    try {
+      Field field =
+          Field.newBuilder()
+              .setSearch(true)
+              .setIndexPrefixes(IndexPrefixes.newBuilder().setMinChars(5).setMaxChars(3).build())
+              .build();
+      createFieldDef(field);
+      fail("Should throw IllegalArgumentException for min_chars > max_chars");
+    } catch (IllegalArgumentException e) {
+      assertEquals("min_chars [5] must be less than max_chars [3]", e.getMessage());
+    }
+  }
+
+  @Test
+  public void testAcceptLength() {
+    Field field =
+        Field.newBuilder()
+            .setSearch(true)
+            .setIndexPrefixes(IndexPrefixes.newBuilder().setMinChars(3).setMaxChars(5).build())
+            .build();
+    TextFieldDef FieldDef = createFieldDef(field);
+    PrefixFieldDef prefixFieldDef = FieldDef.getPrefixFieldDef();
+    assertEquals(3, FieldDef.getPrefixFieldDef().getMinChars());
+    assertEquals(5, FieldDef.getPrefixFieldDef().getMaxChars());
+    assertFalse(prefixFieldDef.accept(0));
+    assertFalse(prefixFieldDef.accept(1));
+    assertTrue(prefixFieldDef.accept(2));
+    assertTrue(prefixFieldDef.accept(5));
+    assertFalse(prefixFieldDef.accept(6));
+  }
+
+  @Test
+  public void testAnalyzer() {
+    Field field =
+        Field.newBuilder()
+            .setSearch(true)
+            .setIndexPrefixes(IndexPrefixes.newBuilder().build())
+            .build();
+    TextFieldDef FieldDef = createFieldDef(field);
+    PrefixFieldDef prefixFieldDef = FieldDef.getPrefixFieldDef();
+    Optional<Analyzer> analyzer = prefixFieldDef.getIndexAnalyzer();
+    assertTrue(analyzer.isPresent());
+    assertTrue(analyzer.get() instanceof PrefixWrappedAnalyzer);
+  }
+
+  @Test
+  public void testPrefixQueryConstantScoreQuery() {
+    Field field =
+        Field.newBuilder()
+            .setSearch(true)
+            .setIndexPrefixes(IndexPrefixes.newBuilder().setMinChars(2).setMaxChars(5).build())
+            .build();
+    TextFieldDef fieldDef = createFieldDef(field);
+
+    PrefixQuery prefixQuery =
+        PrefixQuery.newBuilder().setField("test_field").setPrefix("test").build();
+    Query query = fieldDef.getPrefixQuery(prefixQuery, null);
+    assertNotNull(query);
+    assertTrue(query instanceof ConstantScoreQuery);
+  }
+
+  @Test
+  public void testPrefixQueryBooleanQuery() {
```

Comment:
It would be good to also verify the queries built at the `TextFieldDef` level for all different possibilities. i.e., << min, min - 1, between min and max, > max